core/src/worker/workflow/mod.rs

```diff
@@ -261,10 +262,12 @@ impl Workflows {
             activity_tasks_handle,
             wft_semaphore,
             local_act_mgr,
+            ever_polled: AtomicBool::new(false),
         }
     }
 
     pub(super) async fn next_workflow_activation(&self) -> Result<WorkflowActivation, PollWfError> {
+        self.ever_polled.store(true, atomic::Ordering::Release);
```

Comment:
Probably simpler to just have a worker option that tells the worker that it's an activity only worker.

Reply:
Replace `no_remote_activities` with something like `mode: All | Workflows | Actvities`

Reply:
We know this ahead of time since workflows are registered before worker construction in newer SDKs.

Reply:
That can be screwed up, though. I'd have to return a new error type from workflow polling or something like "You messed up and said you wouldn't do this".

I like this, it's simple, effective, and nothing new for lang.

Reply:
I'm fine with your approach if you prefer it this way.